### ISSUE TITLE
File dragging in the Solution Explorer makes nested nested nested folders

### ISSUE THREAD

**Issue by Lavinski** [Microsoft/visualfsharp#4712]
File dragging is very sensitive and when files or folders are dropped on themselves, they are renamed or copied.

#### Repro steps

![fsharpvisualstudiobug](https://user-images.githubusercontent.com/1761789/38587787-214b4db8-3d66-11e8-8978-64bb7c120af7.gif)

#### Expected behavior

No change to the solution should occur if the file is not moved.

#### Actual behavior

Files are renamed
Folders are copied into themselves.

#### Known workarounds

Be really careful not to drag things in the solution explorer

#### Related information 

Provide any related information 

VS 15.6.2
Microsoft Visual F# Tools 10.1 for F# 4.1

**Comment by cartermp** [comment]
@lifengl is this a CPS issue? I don't think any of the work we did for file ordering affects drag-and-drop.

**Comment by Pilchie** [comment]
Also tagging @TIHan as there *was* definitely some overlap.

**Comment by TIHan** [comment]
@Lavinski Are you using a legacy project?

**Comment by TIHan** [comment]
I can confirm this only happens on the F# legacy project system. In SDK style projects, this is a non-issue as it will not allow it.

**Comment by Lavinski** [comment]
My project has the following:
`<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">`

If this related to dotnetcore or can I switch to the non legacy system. I should note that these projects are recently created. How do you create non legacy projects?

**Comment by TIHan** [comment]
You can switch to a non legacy system.

To create a sdk style project today from the templates, you choose .NET Core (Console Application/Library) or .NET Standard Library. If you need to compile for different targets, for example, .NET Framework 4.6.2, change the ```TargetFramework``` to ```net462``` - ```<TargetFramework>net462</TargetFramework>```.

**Comment by Lavinski** [comment]
Sure, that makes sense. 
However, it's not very clear this is the best way to target net46 from the new project dialog.

**Comment by cartermp** [comment]
@Lavinski Agreed that it's not clear in File | New. The problem is that there are two project systems:

* Legacy/Classic, which only supports .NET Framework and old-style project files
* New, which supports .NET Core/Standard/Framework via only new-style project files

In general, we recommend the new project system for new things and are working towards having it open up old-style projects as well. The legacy project system is complex, slow, and based on an older technology (hence the focus on this new project system).

File | New is something we're focusing on as a division, since it's not an ideal place to have concerns like this represented, let alone for newcomers to VS. We could possibly have templates for .NET Framework via the .NET SDK in File | New in the interim
